### DISCUSSION TITLE
How does deepmd use MPI?

### DISCUSSION THREAD

**Discussion by zhixia721** [deepmodeling/deepmd-kit#9] [Q&A] [answered]
I tried:
mpirun -np 4 do_train water.json
However, it looks the code does not run correctly. I got repeated run information. Do you have any suggestions about how to use mpirun? 
And how about the scaling? Do you have any benchmark?

Thank you!

**Comment by amcadmus** [comment, marked as answer]
For MD simulation: the MPI parallelization is supported. Simply compile lammps with the MPI support (typically compile with "make mpi"), then you can run MPI parallelized deepmd simulations. More details can be found in 
installation: https://github.com/deepmodeling/deepmd-kit#install-lammpss-deepmd-kit-module
usage: https://github.com/deepmodeling/deepmd-kit#run-md-with-lammps.

For training: The current version of deepmd-kit only supports multi-threading parallelization, this should be enough for training with deepmd-kit on one node of a cluster. The distributed parallelization will come with the next release.  The training accelerated by GPU is highly recommended.

**Reply by bose045** [reply]
there is a lmp executable at deepmd_conda_environment/bin/. when I use this lmp executable, it only uses 1 processor, irrespective of how many processor I command to use in mpirun. I suppose that this lmp is a serial lmp. Is there a way to compile am mpi version of lmp while installing deepmd with conda?

**Reply by njzjz** [reply]
All `lmp` executables are built with MPI. You can execute `lmp -h` to print MPI information.
